File: src/main/java/fr/spoutnik87/musicbot_rest/model/MediaType.java
package fr.spoutnik87.musicbot_rest.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Set;

@NoArgsConstructor
@RequiredArgsConstructor
@Data
@Entity
@Table(name = "MediaType")
public class MediaType extends AuditModel implements Serializable {

  @JsonView(Views.Public.class)
  @NonNull
  @Column(nullable = false)
  private String name;

  @JsonView(Views.Public.class)
  @OneToMany(mappedBy = "mediaType", cascade = CascadeType.ALL)
  @JsonBackReference
  private Set<Media> mediaSet;
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/model/MediaType.kt
package fr.spoutnik87.musicbot_rest.model

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonView
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "MediaType")
data class MediaType(
        @JsonView(Views.Companion.Public::class)
        @JsonProperty("id")
        @Column(nullable = false, unique = true)
        var uuid: String,
        @JsonView(Views.Companion.Public::class)
        @Column(nullable = false, unique = true)
        var value: String
) : AuditModel(), Serializable {

    @JsonView(Views.Companion.Public::class)
    @OneToMany(mappedBy = "mediaType", cascade = [CascadeType.ALL])
    @JsonBackReference
    val mediaSet: MutableSet<Media> = HashSet()
}